### PR TITLE
Add a max pending job option

### PR DESCRIPTION
**Pull request by mdebony**
Hello,
I've added an option to limit the job submission in function of the number of pending job.
I think this option could be useful to other people.

The option name is --max-jobs-pending with a default value of 500.

### CODE REVIEW

File: lib/SlurmDagman/process/worker.py [hunk 1]
```diff
@@ -297,9 +305,12 @@ def __write_dag_file(self, dag_file, add_done_labels=True):
 
     def __submit_ready_nodes(self):
         num_submitted_nodes = 0
+        num_nodes_running, num_nodes_pending, num_nodes_unknown = self.__monitor()
```
Comment (AndresTanasijczuk) [inline lib/SlurmDagman/process/worker.py]: I suggest not to call `__monitor()` here again.
`__monitor()` is called in `__execute_dag()` before calling `__submit_ready_nodes()`. So you have `num_nodes_pending` already available and you can pass it as an argument to  `__submit_ready_nodes()`.

**Review by AndresTanasijczuk — CHANGES_REQUESTED** [review]

File: lib/SlurmDagman/config/defaults/common.py [hunk 2]
```diff
@@ -7,5 +7,6 @@
 DEFAULTS['DAGMAN'] = OrderedDict()
 DEFAULTS['DAGMAN']['sleep_time'] = '120'
 DEFAULTS['DAGMAN']['max_jobs_queued'] = '500'
+DEFAULTS['DAGMAN']['max_jobs_pending'] = '500'
```
Comment (AndresTanasijczuk) [inline lib/SlurmDagman/config/defaults/common.py]: I suggest to set a smaller default value for `max_jobs_pending`.
The default for `max_jobs_queued` is 500, where queued corresponds to all transient job states (basically PENDING and RUNNING). Thus, the default for `max_jobs_pending` should be smaller than the default for `max_jobs_queued`.

**Review by AndresTanasijczuk — CHANGES_REQUESTED** [review]

File: lib/SlurmDagman/process/worker.py (hunk 1 above)
Comment (mdebony) [reply lib/SlurmDagman/process/worker.py]: Ok, I will make the modification and test it

File: lib/SlurmDagman/config/defaults/common.py (hunk 2 above)
Comment (mdebony) [reply lib/SlurmDagman/config/defaults/common.py]: I've set it to 500 to be sure to not modified the default behavior of the tool (as with only default value the max_jobs_queued will always be the limiting factor).
Is a value of 100 seem fine to you ?

Comment (AndresTanasijczuk) [reply lib/SlurmDagman/config/defaults/common.py]: I see. Then it is better to set it to 0 than to 500.

**Comment by AndresTanasijczuk** [comment]
Thanks @mdebony for the contribution!

**Pull request merged**